aggregator/src/aggregation/decoder/witgen.rs

```diff
@@ -810,466 +500,217 @@ fn process_block_zstd<F: Field>(
             branch,
             sf_max as u64,
         ],
-        fse_aux_table,
-        huffman_weights,
+        fse_aux_tables,
     )
 }
 
-type LiteralsHeaderProcessingResult<F> = (
+type SequencesProcessingResult<F> = (
     usize,
     Vec<ZstdWitnessRow<F>>,
-    BlockType,
-    usize,
-    usize,
-    usize,
-    (u64, bool),
+    [FseAuxiliaryTableData; 3], // LLT, MLT, CMOT
+    Vec<AddressTableRow>,
 );
 
-fn process_block_zstd_literals_header<F: Field>(
+fn process_sequences<F: Field>(
     src: &[u8],
     byte_offset: usize,
+    end_offset: usize,
     last_row: &ZstdWitnessRow<F>,
     randomness: Value<F>,
-) -> LiteralsHeaderProcessingResult<F> {
-    let lh_bytes = src
-        .iter()
-        .skip(byte_offset)
-        .take(N_MAX_LITERAL_HEADER_BYTES)
-        .cloned()
-        .collect::<Vec<u8>>();
+) -> SequencesProcessingResult<F> {
+    // Initialize witness rows
+    let mut witness_rows: Vec<ZstdWitnessRow<F>> = vec![];
 
-    let literals_block_type = BlockType::from(lh_bytes[0] & 0x3);
-    let size_format = (lh_bytes[0] >> 2) & 3;
-    let sf_max = size_format == 3;
+    // Other consistent values
+    let encoded_len = last_row.encoded_data.encoded_len;
+    let decoded_data = last_row.decoded_data.clone();
 
-    let [n_bits_fmt, n_bits_regen, n_bits_compressed, n_streams, n_bytes_header, branch]: [usize;
-        6] = match literals_block_type {
-        BlockType::RawBlock | BlockType::RleBlock => match size_format {
-            0b00 | 0b10 => [1, 5, 0, 1, 1, 0],
-            0b01 => [2, 12, 0, 1, 2, 1],
-            0b11 => [2, 20, 0, 1, 3, 2],
-            _ => unreachable!("size_format out of bound"),
-        },
-        BlockType::ZstdCompressedBlock => match size_format {
-            0b00 => [2, 10, 10, 1, 3, 3],
-            0b01 => [2, 10, 10, 4, 3, 3],
-            0b10 => [2, 14, 14, 4, 4, 4],
-            0b11 => [2, 18, 18, 4, 5, 5],
-            _ => unreachable!("size_format out of bound"),
-        },
-        _ => unreachable!("BlockType::Reserved unexpected or treeless literal section"),
+    // First, process the sequence header
+    let byte0 = src
+        .get(byte_offset)
+        .expect("First byte of sequence header must exist.")
+        .clone();
+    assert!(byte0 > 0u8, "Sequences can't be of 0 length");
```

Comment:
I think for now this is OK. But this is not true.

[Reference](https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#sequences_section_header):
```
if (Number_of_Sequences == 0) : there are no sequences. The sequence section stops immediately, FSE tables used in Repeat_Mode aren't updated. Block's decompressed content is defined solely by the Literals Section content.
```